core/common/src/main/java/alluxio/wire/CommonOptions.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.wire;
+
+import alluxio.annotation.PublicApi;
+import alluxio.thrift.FileSystemMasterCommonTOptions;
+
+import com.google.common.base.Objects;
+
+import java.io.Serializable;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Common method options.
+ */
+@PublicApi
+@NotThreadSafe
+public final class CommonOptions implements Serializable {
+  private static final long serialVersionUID = -1491370184123698287L;
+
+  private long mSyncIntervalMs;
+
+  /**
+   * @return the default {@link CommonOptions}
+   */
+  public static CommonOptions defaults() {
+    return new CommonOptions();
+  }
+
+  protected CommonOptions() {
+    mSyncIntervalMs = -1;
+  }
+
+  /**
+   * Creates a new instance of {@link CommonOptions} from {@link FileSystemMasterCommonTOptions}.
+   *
+   * @param options Thrift options
+   */
+  public CommonOptions(FileSystemMasterCommonTOptions options) {
+    this();
```

Comment:
do we need to call this?

Reply:
calling `this()` is the way I am setting the defaults, before possibly overwriting them with the input.